backend/lib/endpoints/posts.js

```diff
@@ -31,14 +31,19 @@ async function routes(app) {
   app.get(
     "/",
     {
-      preValidation: [app.authenticate],
+      preValidation: [app.authenticateOptional],
       schema: getPostsSchema,
     },
     async (req) => {
-      const { userId } = req.query;
-      const [userErr, user] = await app.to(User.findById(userId));
-      if (userErr) {
-        throw app.httpErrors.notFound();
+      // TODO: handle optional user if authenticated
+      const { userId } = req;
+      let user;
+      let userErr;
+      if (userId) {
+        [userErr, user] = await app.to(User.findById(userId));
+        if (userErr) {
+          throw app.httpErrors.forbidden();
+        }
       }
```

Comment:
@Naraujo13  Well this was just testing the implementation where you need it, and i figured you could finish this when merging
You could just overwrite this file with your version when rebasing or merging in fact...

Reply:
Also, note that I fixed the error response because it was returning a `404` not found for the `userId` param which should be in encoded in the JWT so that would be more a `403` or something like that